### PR TITLE
add support for AMD brand string

### PR DESCRIPTION
**Pull request by serban300**
Issue #, if available: #1011 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: arch/src/x86_64/regs.rs [hunk 1]
```diff
@@ -441,6 +441,6 @@ mod tests {
 
         setup_sregs(&gm, &vcpu).unwrap();
         let actual_sregs: kvm_sregs = vcpu.get_sregs().unwrap();
-        assert_eq!(expected_sregs, actual_sregs);
+        //        assert_eq!(expected_sregs, actual_sregs);
```
Comment (dianpopa) [inline arch/src/x86_64/regs.rs]: Should this remain a comment?

File: cpuid/src/brand_string.rs [hunk 2]
```diff
@@ -66,6 +70,37 @@ impl BrandString {
...
     }
 
+    /// Generates the emulated brand string.
+    ///
+    /// For Intel CPUs, the brand string we expose will be:
+    ///    "Intel(R) Xeon(R) Processor @ {host freq}"
+    /// where {host freq} is the CPU frequency, as present in the
+    /// host brand string (e.g. 4.01GHz).
+    ///
+    /// For AMD CPUs, the brand string we expose will be AMD EPYC
```
Comment (dianpopa) [inline cpuid/src/brand_string.rs]: nit: full stop

File: cpuid/src/brand_string.rs [hunk 3]
```diff
@@ -340,6 +362,12 @@ mod tests {
                 "This function should not return another type of error"
             ),
         }
+
+        // Test BrandString::from_vendor_id()
+        let bstr = BrandString::from_vendor_id(VENDOR_ID_INTEL);
+        assert!(bstr.as_bytes().starts_with(BRAND_STRING_INTEL));
+        let bstr = BrandString::from_vendor_id(VENDOR_ID_AMD);
+        assert!(bstr.as_bytes().starts_with(BRAND_STRING_AMD));
```
Comment (dianpopa) [inline cpuid/src/brand_string.rs]: Let s add a test for this branch also: "For other CPUs, we'll just expose an empty string.".

File: cpuid/src/brand_string.rs [hunk 4]
```diff
@@ -66,6 +70,37 @@ impl BrandString {
...
+    /// For Intel CPUs, the brand string we expose will be:
+    ///    "Intel(R) Xeon(R) Processor @ {host freq}"
+    /// where {host freq} is the CPU frequency, as present in the
+    /// host brand string (e.g. 4.01GHz).
+    ///
+    /// For AMD CPUs, the brand string we expose will be AMD EPYC
+    ///
+    /// For other CPUs, we'll just expose an empty string.
+    ///
+    /// This is safe because we know DEFAULT_BRAND_STRING to hold valid data
```
Comment (dianpopa) [inline cpuid/src/brand_string.rs]: Does DEFAULT_BRAND_STRING exist anymore?

File: arch/src/x86_64/regs.rs (hunk 1 above)
Comment (serban300) [reply arch/src/x86_64/regs.rs]: Oops. Yes. Fixed.

File: cpuid/src/brand_string.rs (hunk 2 above)
Comment (serban300) [reply cpuid/src/brand_string.rs]: Done

File: cpuid/src/brand_string.rs (hunk 4 above)
Comment (serban300) [reply cpuid/src/brand_string.rs]: No. Fixed.

File: cpuid/src/brand_string.rs (hunk 3 above)
Comment (serban300) [reply cpuid/src/brand_string.rs]: Done

File: cpuid/src/brand_string.rs [hunk 5]
```diff
@@ -288,10 +315,6 @@ mod tests {
 
         // Test BrandString::starts_with()
```
Comment (dianpopa) [inline cpuid/src/brand_string.rs]: nit: remove leftover comment

Comment (serban300) [reply cpuid/src/brand_string.rs]: Done
